src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -262,6 +262,15 @@ def create_managed_disk(cmd, resource_group_name, disk_name, location=None,  # p
     else:
         option = DiskCreateOption.empty
 
+    if source_storage_account_id is None and source_blob_uri is not None:
+        from azure.cli.core.commands.client_factory import get_subscription_id
+        from msrestazure.tools import resource_id
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        storage_account_name = source_blob_uri.split('.')[0].split('/')[-1]
+        source_storage_account_id = resource_id(
+            subscription=subscription_id, resource_group=resource_group_name,
```

Comment:
The subscription ID of the Storage Account may not be the current one. Per [doc](https://docs.microsoft.com/en-us/cli/azure/disk?view=azure-cli-latest#az-disk-create)
```
--source-storage-account-id
Used when source blob is in a different subscription.
```

Reply:
User needs to specify it if it's in a different subscription.